odoo/addons/base/models/res_users.py

```diff
@@ -1618,6 +1619,27 @@ def remove(self):
             return {'type': 'ir.actions.act_window_close'}
         raise AccessError(_("You can not remove API keys unless they're yours or you are a system user"))
 
+    def get_api_key(self, key):
+        index = key[:INDEX_SIZE]
+        self.env.cr.execute('SELECT id, key FROM res_users_apikeys WHERE index = %s', [index])
+        for [id, current_key] in self.env.cr.fetchall():
+            if KEY_CRYPT_CONTEXT.verify(key, current_key):
+                return self.browse(id)
```

Comment:
Maybe return an empty recordset on failure instead of `None`? Not sure there's a point to returing a key though, unless a `scope` field gets added it won't be accessible.

Reply:
I intended to use it like this:

```python
    @classmethod
    def _auth_method_oauth_outlook(cls):
        api_token = request.httprequest.headers.get('Token')
        if not api_token:
            return False

        key_identity= request.env["res.users.apikeys"].get_api_key(api_token)
        if not key_identity:
            return False

        if key_identity.scope and key_identity.scope != 'outlook':
            return False

        # take the identity of the API key user
        request.uid = key_identity.sudo().user_id.id
        print("API logged as", key_identity.sudo().user_id.name)
        return True
```

But I could just return the `user_id` and the `scope`...

Reply:
Alternatively, I could also pass the scope to the _get_api_key/_check_credentials, like ODO suggests

```python
    def _user_for_key(self, key, scope=None):
        """Verify key and returns the owner user ID"""
        # (...)
        for (user_id, user_key) in self.env.cr.fetchall():
           if KEY_CRYPT_CONTEXT.verify(key, user_key):
               return user_id
        return None
```

Reply:
> I intended to use it like this:

I understand that, but `scope` is *not a field*, it's only accessible over SQL, so that wouldn't work.